csharp/src/Microsoft.ML.OnnxRuntime/ProviderOptions.shared.cs

```diff
@@ -318,9 +318,9 @@ public static void StringToDict(string s, Dictionary<string, string> dict)
     }
 
     /// <summary>
-    /// CoreML flags for use with SessionOptions
+    /// CoreML flags for use with SessionOptions.
+    /// See https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/providers/coreml/coreml_provider_factory.h
     /// </summary>
-    /// <see cref="https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/providers/coreml/coreml_provider_factory.h"/>
```

Comment:
`cref` generates a warning as this url is an invalid valid for a cref
`<see href=...` would be nicer and is valid for xmldoc, but we generate docs with doxygen and it doesn't allow it.